src/server/repositories/UrlRepository.ts

```diff
@@ -54,28 +58,57 @@ export class UrlRepository implements UrlRepositoryInterface {
   }
 
   public create: (
-    properties: { userId: number; shortUrl: string; longUrl?: string },
+    properties: {
+      userId: number
+      shortUrl: string
+      longUrl?: string
+      tags?: string[]
+    },
     file?: StorableFile,
   ) => Promise<StorableUrl> = async (properties, file) => {
     const newUrl = await sequelize.transaction(async (t) => {
-      await Url.create(
-        {
-          ...properties,
-          longUrl: file
-            ? this.fileBucket.buildFileLongUrl(file.key)
-            : properties.longUrl,
-          isFile: !!file,
-        },
-        {
-          transaction: t,
-        },
-      )
+      const tagStrings = properties.tags
+        ? properties.tags.join(tagSeparator)
+        : ''
+      const urlStaticDTO = {
+        ...properties,
+        longUrl: file
+          ? this.fileBucket.buildFileLongUrl(file.key)
+          : properties.longUrl,
+        isFile: !!file,
+        tagStrings,
+      }
+      const url = await Url.create(urlStaticDTO, {
+        transaction: t,
+      })
+      if (properties.tags) {
+        const tagCreationResponses = await Promise.all(
+          properties.tags.map(async (tag) => {
+            return Tag.findOrCreate({
+              where: {
+                tagString: tag,
+                tagKey: tag.toLowerCase(),
+              },
+              transaction: t,
+            })
+          }),
+        )
+        await Promise.all(
+          tagCreationResponses.map(async (response) => {
+            const [tag, _] = response
+            if (tag) {
+              // @ts-ignore, addTag is provided by Sequelize during run time.
+              await url.addTag(tag, { transaction: t })
```

Comment:
I'm not very familiar with this Sequelize special methods/mixins concept, but I'm wondering if we can just call `setTags` instead of calling `addTag` one by one?

Reply:
Suggestion: can we also add the [url to the documentation](https://sequelize.org/docs/v6/core-concepts/assocs/#special-methodsmixins-added-to-instances) for the special methods inline where we use `addTag` and `setTags` so that a future developer looking at this can look directly at the documentation as well?

Reply:
addTags is actually better, will do this as well

Reply:
done